### PR TITLE
More post v2 cleanup

### PR DESCRIPTION
**Pull request by mattiecnvr**
### Description

_What's included in this PR?_
* Deleted an old inbox file that was not part of the package structure
* Renamed outbox/inbox in strings
* Renamed outbox/inbox in comments/docs
* Removed unused outbox state metric

### Drive-by changes

_Are there any minor or drive-by changes also included?_

### Backward compatibility

_Are these changes backward compatible?_

Yes

_Are there any infrastructure implications, e.g. changes that would prohibit deploying older commits using this infra tooling?_

None

### Testing

_What kind of testing have these changes undergone?_

Manual
Unit Tests

### CODE REVIEW

File: rust/agents/relayer/src/msg/processor.rs [hunk 1]
```diff
@@ -118,7 +118,7 @@ impl MessageProcessor {
             return Ok(());
         };
 
-        // Skip if for different inbox.
+        // Skip if for different mailbox.
```
Comment (asaj) [inline rust/agents/relayer/src/msg/processor.rs:121]: nit: Consider "skip if for different domain"

File: rust/agents/relayer/src/msg/serial_submitter.rs [hunk 2]
```diff
@@ -237,16 +237,16 @@ impl SerialSubmitter {
...
-    /// in this fn or by a view call to the Inbox contract discovering the message has already
+    /// in this fn or by a view call to the Mailbox contract discovering the message has already
     /// been processed, Ok(MessageStatus::Processed) is returned. If this message is unable to
     /// be processed, either due to failed gas estimation or an insufficient gas payment,
     /// Ok(MessageStatus::None) is returned.
     #[instrument(skip(self, msg), fields(msg_nonce=msg.message.nonce))]
     async fn process_message(&self, msg: &SubmitMessageArgs) -> Result<bool> {
         // If the message has already been processed according to message_status call on
-        // inbox, e.g. due to another relayer having already processed, then mark it as
+        // mailbox, e.g. due to another relayer having already processed, then mark it as
```
Comment (asaj) [inline rust/agents/relayer/src/msg/serial_submitter.rs:247]: nit: "message_status" is outdated

File: rust/agents/relayer/src/msg/serial_submitter.rs [hunk 3]
```diff
@@ -237,16 +237,16 @@ impl SerialSubmitter {
     }
 
     /// Returns the message's status. If the message is processed, either by a transaction
-    /// in this fn or by a view call to the Inbox contract discovering the message has already
+    /// in this fn or by a view call to the Mailbox contract discovering the message has already
```
Comment (asaj) [inline rust/agents/relayer/src/msg/serial_submitter.rs:240]: "MessageStatus::Processed/None" is outdated

File: rust/agents/scraper/src/db/message.rs [hunk 4]
```diff
@@ -55,7 +55,7 @@ impl ScraperDb {
             .map(|idx| idx as u32))
     }
 
-    /// Store deliveries from an inbox into the database (or update an existing
+    /// Store deliveries from an mailbox into the database (or update an existing
```
Comment (asaj) [inline rust/agents/scraper/src/db/message.rs:58]: nit: an/a

File: rust/agents/scraper/src/db/message.rs [hunk 5]
```diff
@@ -96,7 +96,7 @@ impl ScraperDb {
         Ok(())
     }
 
-    /// Store messages from an outbox into the database (or update an existing
+    /// Store messages from an mailbox into the database (or update an existing
```
Comment (asaj) [inline rust/agents/scraper/src/db/message.rs:99]: ditto

File: rust/hyperlane-base/src/settings/mod.rs [hunk 6]
```diff
@@ -5,7 +5,7 @@
 //! Hyperlane Agents have a shared core, which contains connection info for rpc,
 //! relevant contract addresses on each chain, etc. In addition, each agent has
 //! agent-specific settings. Be convention, we represent these as a base config
-//! per-Outbox contract, and a "partial" config per agent. On bootup, the agent
+//! per-Mailbox contract, and a "partial" config per agent. On bootup, the agent
```
Comment (asaj) [inline rust/hyperlane-base/src/settings/mod.rs:8]: nit: "By convention above

File: rust/hyperlane-core/src/traits/indexer.rs [hunk 7]
```diff
@@ -3,7 +3,7 @@
 //! way to retrieve data such as the chain's latest block number or a list of
 //! checkpoints/messages emitted within a certain block range by calling out to
 //! a chain-specific library and provider (e.g. ethers::provider). A
-//! chain-specific mailbox or inbox should implement one or both of the Indexer
+//! chain-specific mailbox should implement one or both of the Indexer
```
Comment (asaj) [inline rust/hyperlane-core/src/traits/indexer.rs:6]: Is CommonIndexer still a thing?

**Review by asaj — COMMENTED** [review]
Thanks! A few nits

File: rust/hyperlane-core/src/traits/indexer.rs (hunk 7 above)
Comment (mattiecnvr) [reply rust/hyperlane-core/src/traits/indexer.rs]: No, it is not

**Review by mattiecnvr — COMMENTED** [review]
Made changes as requested

**Pull request merged**